### ISSUE TITLE
Some of the buttons lack valid visual indication of focus due to poor color contrast

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#4667]
[Clarity Version]
3.1.1

[Test URL]
https://clarity.design/documentation/buttons#examples

[WCAG 2.1 Guideline]
Success Criterion 2.4.3 Focus Order
https://www.w3.org/TR/WCAG21/#focus-order
https://www.w3.org/WAI/WCAG21/Techniques/failures/F78
(Level A)

[OS]
Windows 10

[User Agents/Assistive Technologies]
Firefox 75.0, Switch

[Component]
Buttons

[Description]
Some of the buttons lack valid visual indication of focus due to poor color contrast. This occurs in many examples that contain buttons. 	

[User Impact]
Switch users have difficulty tracking their keyboard focus because they cannot visually see that their focus is on the button. This also makes it difficult to activate the button because they cannot see that their focus is on the button

[Steps to Reproduce]
See the following instructions for switch usage then follow the steps below.

Windows (see Scan through keys)
https://support.microsoft.com/en-us/help/10762/windows-use-on-screen-keyboard

1. Using a switch, try navigating to and activating the Primary button under the Solid Buttons heading. 
2. Notice that there is no visual indication of focus in order to know which button you are on.

Mac
https://support.apple.com/en-us/HT202865

[Recommendation]
The current outline rules contain a focus ring for webkit browsers, and a standard dotted outline for other browsers. However, in Firefox, when the color of an outline is not specified the default color of the button is used. In this case, it creates a white outline on the light grey background. In order to ensure an outline with sufficient color contrast, an outline-offset should be added with a negative value. This will push the outline into the boundary of the button, ensuring that sufficiently visible focus indicator is provided. The outline-offset property will not impact the webkit focus ring.	button:focus {    outline: 1px dotted;    outline: 5px auto -webkit-focus-ring-color;    outline-offset: -3px;}

*chlane - gkhadke found that when the user agent's default focus indicator is obscured by the styling of another element, the author needs to supply a custom visual focus indication or remove the styling that is obscuring the visual focus indication. See https://www.w3.org/WAI/WCAG21/Techniques/failures/F78 found that when the user agent&#039;s default focus indicator is obscured by the styling of another element, the author needs to supply a custom visual focus indication or remove the styling that is obscuring the visual focus indication. See https://www.w3.org/WAI/WCAG21/Techniques/failures/F78

[Reporter Email] 
daman@damanw.com

**Comment by chlane** [comment]
See https://confluence.eng.vmware.com/display/A11Y/Visual+Focus+Indication.

I recommend the following technique for visual focus indication since it can adapt to different backgrounds and Windows High Contrast modes.



*:focus {

outline: 4px solid blue;

box-shadow: 0 0 0 1px white, 0 0 0 4px black, 0 0 0 4px white;

z-index: 1;

}
